### PR TITLE
Preserve pigen image and container

### PR DESCRIPTION
**Pull request by jredrejo**
### Summary
Our building system prune all the docker images and containers after finishing.
When building the Raspberry Pi image, if the docker container is not kept, all the Debian system and packages are downloaded and installed once again in every run of the builder.
This PR keeps undeleted the image and container for the Raspberry Pi image. They should be deleted only when a new Debian distribuition for Raspberry is released, what happen every 2 years aproximately.

### Reviewer guidance
The pigen_work container and pi-gen images still exists in the building server after the image is built. 

### Contributor Checklist

PR process:

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

**Review by DXCanas — APPROVED** [review]
This will work, but we need something a little more sustainable down the line. 

I still thinking blocking the build by default would be the way to go, as in #https://github.com/learningequality/kolibri/pull/6402

**Comment by indirectlylit** [comment]
should this be targeted to 0.13?

**Comment by DXCanas** [comment]
@indirectlylit I think that's where it'd be most helpful, yeah

**Comment by indirectlylit** [comment]
needs a rebase

**Comment by jredrejo** [comment]
I have changed the code to use actual labels that are applied in https://github.com/learningequality/pi-gen/blob/master/Dockerfile#L2
Previous method did not work because neither the container nor the image were labeled. It seems Docker does not consider the image name a label nor provide options to filter by the image or container names.

**Comment by lyw07** [comment]
for the current build failure, I think https://github.com/learningequality/kolibri/pull/6402 can fix the issue?

**Comment by DXCanas** [comment]
@lyw07 #6402 was only helpful if we want to block the build from buildking every time. I'm thinking we should maybe hold off on doing that until after release. This build should pass with the latest changes, though.

**Comment by lyw07** [comment]
thank you for the info! I will merge this PR after the build passes

**Pull request merged**
